umap/static/umap/js/modules/alerts.js

```diff
@@ -0,0 +1,54 @@
+export default class Alerts {
+  constructor() {
+    this.alertNode = document.querySelector('[role="alert"]')
+    const observer = new MutationObserver(this._callback.bind(this))
+    observer.observe(this.alertNode, { childList: true })
+    // On initial page load, we want to display messages from Django.
+    Array.from(this.alertNode.children).forEach(this._display.bind(this))
+  }
+
+  _callback(mutationList, observer) {
+    for (const mutation of mutationList) {
+      this._display(
+        [...mutation.addedNodes].filter((item) => item.tagName === 'P').pop()
+      )
```

Comment:
Why do we need to `.pop()` the element if we are already filtering it? 🤔

Reply:
`filter()` might return multiple element and we want the first one.

Reply:
Thanks, I actually misread, and was trying to understand what was being poped-out (because it's removed from the list). Turns out it's not a problem here because the whole list only exists for the duration of the event.